stan/math/opencl/prim/add.hpp

```diff
@@ -0,0 +1,29 @@
+#ifndef STAN_MATH_OPENCL_PRIM_ADD_HPP
+#define STAN_MATH_OPENCL_PRIM_ADD_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/opencl/matrix_cl.hpp>
+#include <stan/math/opencl/kernel_generator.hpp>
+
+namespace stan {
+namespace math {
+/** \ingroup opencl
+ * Computes the sum on kernel generator expressions.
+ *
+ * @param a first expression
+ * @param b second expression
+ * @return the sum of the first and second expression
+ *
+ * @throw <code>std::invalid_argument</code> if the
+ *   dimensions dont match if none of tha arguments are a scalar.
+ */
+template <typename T_a, typename T_b,
+          typename = require_all_valid_expressions_t<T_a, T_b>,
+          typename = require_any_not_arithmetic_t<T_a, T_b>>
```

Comment:
Should `require_all_valid_expressions_t` be named something like `require_all_matrix_cl_expressions_t`? I think that's a nicer name since the idea here is to allow matrix_cl objects or expressions derived from matrix_cl

Reply:
I am going to defer to @t4c1 for this one. I agree with you, but maybe there is some other reason behind the name.

Reply:
What about `require_all_kernel_expressions_t`? I would hesitate to call this matrix expressions as it also accepts scalars.

Reply:
I am cool with that. If @SteveBronder also agrees, I will change it in this PR and update the top level comment/release notes.

Reply:
I'm fine with something like `require_all_kernel_expressions_t` or maybe `require_all_kernelcl_expressions_t`

Reply:
Thanks. Went with require_all_kernel_expressions_t and is_valid_kernel_expression.

Reply:
I think `require_all_kernel_expressions_t` is fine. `is_valid_kernel_expression` should instead be `is_kernel_expression`.